lib/option.fz

```diff
@@ -96,9 +96,17 @@ is
 
 
   # synonym for infix >>=
+  #
   public and_then (f T -> option T) => option.this >>= f
 
 
+  # if this option is nil return the result of f
+  # otherwise just return this option.
+  #
+  public or_then (f () -> option T) option T =>
```

Comment:
Should IMHO be called `or_else`, or just `or`, this is at least the name chosen in Java's `Optional`.  The type of `f` should be `Lazy (option T)` for symmetry with `get`/`or`/`or_else` (whatever the final name will be) below.